token-swap/program/src/curve/stable.rs

```diff
@@ -216,19 +216,19 @@ impl CurveCalculator for StableCurve {
         };
         let leverage = self.amp.checked_mul(N_COINS as u64)?;
         let d0 = compute_d(leverage, swap_source_amount, swap_destination_amount)?;
-        let (new_swap_source_amount, new_swap_destination_amount) = match liquidity_provider_operation {
-            LiquidityProviderOperation::Deposit => (swap_token_a_amount + source_amount, swap_token_b_amount),
-            LiquidityProviderOperation::Withdrawal => (swap_token_b_amount - source_amount, swap_token_a_amount)
+        let new_swap_source_amount = match liquidity_provider_operation {
+            LiquidityProviderOperation::Deposit => swap_source_amount + source_amount,
+            LiquidityProviderOperation::Withdrawal => swap_source_amount - source_amount,
```

Comment:
I forgot to mention this earlier, but this should use checked math, `checked_add` and `checked_sub`